src/events.js

```diff
@@ -119,15 +119,19 @@ class Event {
      * functions.
      */
     notifyListeners(listenerArguments = []) {
+        let returnVal = undefined;
         this.listeners.forEach((options, listener) => {
             try {
-                if((this.notifyListenersCallback === null) || this.notifyListenersCallback(listener, listenerArguments, options))
-                    listener.apply(null, listenerArguments);
+                if((this.notifyListenersCallback === null) || this.notifyListenersCallback(listener, listenerArguments, options)) {
+                    const listenerReturnValue = listener.apply(null, listenerArguments);
+                    if (listenerReturnValue !== undefined) { returnVal = listenerReturnValue; }
+                }
             }
             catch(error) {
                 debugLog(`Error in listener notification: ${error}`);
             }
         });
+        return returnVal;
```

Comment:
Rather than clobbering the return value for each listener, so the `notifyListeners` caller only receives the last return value, what about storing all the return values in an array and returning that to the `notifyListeners` caller?